src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -885,7 +902,7 @@ public void getDocContent(Request req, Response resp) throws IOException {
 
     // TODO(mifern): Include extended attributes.
 
-    if (delegate.isDfsNamespace(doc)) {
+    if (!allowFilesInDfsNamespaces && delegate.isDfsNamespace(doc)) {
```

Comment:
So if allowFiles is true we just treat the DFS namespace like an ordinary directory, modulo the changes in readShareAcls?